test/profiling/MemoryTestPublisher.cpp

```diff
@@ -94,261 +79,293 @@ bool MemoryTestPublisher::init(
+        return false;
     }
 
     return true;
 }
 
-void MemoryTestPublisher::DataPubListener::onPublicationMatched(
-        Publisher* /*pub*/,
-        MatchingInfo& info)
+void MemoryTestPublisher::DataPubListener::on_publication_matched(
+        DataWriter*,
+        const PublicationMatchedStatus& info)
 {
-    std::unique_lock<std::mutex> lock(mp_up->mutex_);
+    std::unique_lock<std::mutex> lock(up_->mutex_);
 
-    if (info.status == MATCHED_MATCHING)
+    if (0 < info.current_count_change)
     {
         cout << C_MAGENTA << "Data Pub Matched " << C_DEF << endl;
 
-        n_matched++;
-        if (n_matched > mp_up->n_subscribers)
+        n_matched = info.total_count;
+        if (n_matched > up_->n_subscribers)
         {
             std::cout << "More matched subscribers than expected" << std::endl;
-            mp_up->m_status = -1;
+            up_->m_status = -1;
         }
 
-        ++mp_up->disc_count_;
     }
     else
     {
         cout << C_MAGENTA << "Data Pub Unmatched " << C_DEF << endl;
-        --mp_up->disc_count_;
     }
+    up_->disc_count_ += info.current_count_change;
 
     lock.unlock();
-    mp_up->disc_cond_.notify_one();
+    up_->disc_cond_.notify_one();
 }
 
-void MemoryTestPublisher::CommandPubListener::onPublicationMatched(
-        Publisher* /*pub*/,
-        MatchingInfo& info)
+void MemoryTestPublisher::CommandPubListener::on_publication_matched(
+        DataWriter*,
+        const PublicationMatchedStatus& info)
 {
-    std::unique_lock<std::mutex> lock(mp_up->mutex_);
+    std::unique_lock<std::mutex> lock(up_->mutex_);
 
-    if (info.status == MATCHED_MATCHING)
+    if (0 < info.current_count_change)
     {
         cout << C_MAGENTA << "Command Pub Matched " << C_DEF << endl;
 
-        n_matched++;
-        if (n_matched > mp_up->n_subscribers)
+        n_matched = info.total_count;
+        if (n_matched > up_->n_subscribers)
         {
             std::cout << "More matched subscribers than expected" << std::endl;
-            mp_up->m_status = -1;
+            up_->m_status = -1;
         }
-
-        ++mp_up->disc_count_;
     }
     else
     {
         cout << C_MAGENTA << "Command Pub unmatched " << C_DEF << endl;
-        --mp_up->disc_count_;
     }
+    up_->disc_count_ += info.current_count_change;
 
     lock.unlock();
-    mp_up->disc_cond_.notify_one();
+    up_->disc_cond_.notify_one();
 }
 
-void MemoryTestPublisher::CommandSubListener::onSubscriptionMatched(
-        Subscriber* /*sub*/,
-        MatchingInfo& info)
+void MemoryTestPublisher::CommandSubListener::on_subscription_matched(
+        DataReader*,
+        const SubscriptionMatchedStatus& info)
 {
-    std::unique_lock<std::mutex> lock(mp_up->mutex_);
+    std::unique_lock<std::mutex> lock(up_->mutex_);
 
-    if (info.status == MATCHED_MATCHING)
+    if (0 < info.current_count_change)
     {
         cout << C_MAGENTA << "Command Sub Matched " << C_DEF << endl;
```

Comment:
I suggest removing `using std::cout`/`std::endl` from the file so every use have to determine the namespace.

Reply:
Done in d5af16ae0